scaper/core.py

```diff
@@ -1380,9 +1442,9 @@ def _instantiate_event(self, event, isbackground=False,
             used_labels.append(label)
 
         # determine source file
+        # special case: choose tuple with empty list
         if event.source_file[0] == "choose" and not event.source_file[1]:
```

Comment:
Why isn't this also happening for `choose_weighted`?

Reply:
Oh so the lists of labels and probabilities have to match in length. Got it.

Reply:
same as last comment, it's a special case where we need to update the "choose" tuple with info from disk